lib/mix/phoenix.ex

```diff
@@ -3,13 +3,24 @@ defmodule Mix.Phoenix do
   @moduledoc false
 
   @doc """
-  Returns the module base name based on the application name.
+  Returns the module base name based on the configuration value.
+
+      config :my_app
+        phoenix_namespace: My.App
+
   """
   def base do
-    Mix.Project.config
-    |> Keyword.fetch!(:app)
-    |> to_string
-    |> Phoenix.Naming.camelize
+    app = Mix.Project.config |> Keyword.fetch!(:app)
+
+    Mix.Config.read!("config/config.exs")
```

Comment:
However, if no phoenix namespace is set, we need to use the same value as before, so we do need to keep the previous code! We will also need tests. :)